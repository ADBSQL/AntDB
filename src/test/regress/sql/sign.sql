set grammar to oracle;
--加号	
select 1+1-2 from dual;
select 2+1.78-2.1 from dual;
select 1+'1.0' from dual;
select '1'+'1.0' from dual;
select '1'+'1' from dual;
select 1+to_char(1) from dual;
select to_char(1)+to_char(1.2) from dual;
select 1+to_char(1) from dual;
create table tt(id number,name varchar2(3),ad char(3));
insert into tt values(1,'1.0','1');
select 1+name from tt;
select 1+ad from tt;
drop table tt;
select 1 + '1.0'::varchar2(3) from dual;
select 1 + '1'::char(3) from dual;
select to_number(1)+'1' from dual;
select 1.0::binary_float+'1' from dual;
select 1.0::float(4)+'1' from dual;	
select to_char(to_date('2014-07-15','YYYY-MM-DD')+numtoyminterval(3,'month'),'yyyy-mm') from dual;
select to_char(numtoyminterval(3,'year') - numtoyminterval(3,'month'),'yyyy-mm') from dual;
--减	
select 1-'1' from dual;
select 1-'1.0' from dual;
select '1'-'1.0' from dual;
select '1'-'1' from dual;
select 1-to_char(1) from dual;
select to_char(1)-to_char(1.2) from dual;
select 1-to_char(1) from dual;
select to_number(1)-'1' from dual;
select 1.0::binary_float-'1' from dual;
select 1.0::float(4)-'1' from dual;
--乘			
select 1*'1' from dual;
select 1*'1.0' from dual;
select '1'*'1.0' from dual;
select '1'*'1' from dual;
select 1*to_char(1) from dual;
select to_char(1)*to_char(1.2) from dual;
select 1*to_char(1) from dual;
select 1 *'1.0'::varchar2(3) from dual;
select 1 * '1'::char(3) from dual;
select to_number(1)*'1' from dual;
select 1.0::binary_float*'1' from dual;
select 1.0::float(4)*'1' from dual;
select to_number(1)*'1' from dual;
--除
select to_number(1)/'1' from dual;
select to_number(5)/to_number(2) from dual;
select to_number(5)/'2' from dual;
select 1.0::binary_float/'1' from dual;
select 1.0::float(4)/'1' from dual;
select 1/'1' from dual;
select 1/'1.0' from dual;
select '1'/'1.0' from dual;
select '1'/'1' from dual;
select 1/to_char(1) from dual;
select to_char(1)/to_char(1.2) from dual;
select 1/to_char(1) from dual;
select 1/'1.0'::varchar2(3) from dual;
select 1/'1'::char(3) from dual;
--比较运算符	
create table tt(id number, name varchar2(3));
insert into tt values(1,'1.0');
select * from tt where name = 1;
drop table tt;
select 'equal' from dual where 1>=1.0::varchar(3);
select 'equal' from dual where 1>=to_char(1.0);
select 'equal' from dual where 1>='1.0';
select 'equal' from dual where 1>=1::varchar(3);
select 'equal' from dual where 1>=1::char(3);
create table tt(id number,job varchar(10));
insert into tt values(1,'IT');
insert into tt values(2,'doctor');
insert into tt values(3,'teacher');
insert into tt values(6,'');
create table aa(id char,avgm number);
insert into aa values(1.00,10000);
insert into aa values(2,5000);
insert into aa values(4,9000);
insert into aa values(5,null);
select * from tt where id > all(select id from aa);
drop table tt;
drop table aa;
--between and		
select 1 from dual where 3 between to_char(2) and to_char(5);
select 1 from dual where 3 between 2.0::char(5) and 5.0::varchar(5);
select 1 from dual where 3 between '2.0' and '5.0';
select 1 from dual where to_char(3) between '2.0' and to_char(5);
select 1 from dual where to_char(3) between to_char(2) and to_char(5);
select 1 from dual where 3.0 between '2.0' and '5.0';
select 'equal' from dual where 1=1::number(2);
select 'equal' from dual where 1=1.0::number(3,2);
select 'equal' from dual where 1=1.0::binary_float;
--||	
select 1||2 as result from dual;
select 1.0||2.0 as result from dual;
select to_number(1.20)||to_number(2.0) as result from dual;
select 1.00::binary_double||2.00::binary_double as result from dual;
select 1||to_char('2') as result from dual;
select 1.0||to_char('2') as result from dual;
select to_char(1.0)||to_char(2.0) as result from dual;
select to_char(1.00)||to_char('b') as result from dual;
select 1||'2' as result from dual;
select '1'||'2' as result from dual;
select to_char(1)||to_char(2) as result from dual;
select to_char('1.0')||to_char('2.0') as result from dual;
select to_char('a')||to_char('b') as result from dual;
select to_char(1)||to_char('b') as result from dual;
