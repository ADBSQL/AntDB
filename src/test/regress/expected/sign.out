set grammar to oracle;
--加号	
select 1+1-2 from dual;
 ?COLUMN? 
----------
        0
(1 row)

select 2+1.78-2.1 from dual;
 ?COLUMN? 
----------
     1.68
(1 row)

select 1+'1.0' from dual;
 ?COLUMN? 
----------
        2
(1 row)

select '1'+'1.0' from dual;
 ?COLUMN? 
----------
        2
(1 row)

select '1'+'1' from dual;
 ?COLUMN? 
----------
        2
(1 row)

select 1+to_char(1) from dual;
 ?COLUMN? 
----------
        2
(1 row)

select to_char(1)+to_char(1.2) from dual;
 ?COLUMN? 
----------
      2.2
(1 row)

select 1+to_char(1) from dual;
 ?COLUMN? 
----------
        2
(1 row)

create table signtbl1(id number,name varchar2(3),ad char(3));
insert into signtbl1 values(1,'1.0','1');
select 1+name from signtbl1;
 ?COLUMN? 
----------
        2
(1 row)

select 1+ad from signtbl1;
 ?COLUMN? 
----------
        2
(1 row)

drop table signtbl1;
select 1 + '1.0'::varchar2(3) from dual;
 ?COLUMN? 
----------
        2
(1 row)

select 1 + '1'::char(3) from dual;
 ?COLUMN? 
----------
        2
(1 row)

select to_number(1)+'1' from dual;
 ?COLUMN? 
----------
        2
(1 row)

select 1.0::binary_float+'1' from dual;
 ?COLUMN? 
----------
        2
(1 row)

select 1.0::float(4)+'1' from dual;	
 ?COLUMN? 
----------
        2
(1 row)

select to_char(to_date('2014-07-15','YYYY-MM-DD')+numtoyminterval(3,'month'),'yyyy-mm') from dual;
 TO_CHAR(TO_DATE('2014-07-15','YYYY-MM-DD')+NUMTOYMINTERVAL(3,'MONTH'),'YYYY-MM') 
----------------------------------------------------------------------------------
 2014-10
(1 row)

select to_char(numtoyminterval(3,'year') - numtoyminterval(3,'month'),'yyyy-mm') from dual;
 TO_CHAR(NUMTOYMINTERVAL(3,'YEAR') - NUMTOYMINTERVAL(3,'MONTH'),'YYYY-MM') 
---------------------------------------------------------------------------
 0002-09
(1 row)

--减	
select 1-'1' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select 1-'1.0' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select '1'-'1.0' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select '1'-'1' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select 1-to_char(1) from dual;
 ?COLUMN? 
----------
        0
(1 row)

select to_char(1)-to_char(1.2) from dual;
 ?COLUMN? 
----------
     -0.2
(1 row)

select 1-to_char(1) from dual;
 ?COLUMN? 
----------
        0
(1 row)

select to_number(1)-'1' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select 1.0::binary_float-'1' from dual;
 ?COLUMN? 
----------
        0
(1 row)

select 1.0::float(4)-'1' from dual;
 ?COLUMN? 
----------
        0
(1 row)

--乘			
select 1*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1*'1.0' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select '1'*'1.0' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select '1'*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1*to_char(1) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select to_char(1)*to_char(1.2) from dual;
 ?COLUMN? 
----------
      1.2
(1 row)

select 1*to_char(1) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1 *'1.0'::varchar2(3) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1 * '1'::char(3) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select to_number(1)*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1.0::binary_float*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1.0::float(4)*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select to_number(1)*'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

--除
select to_number(1)/'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select to_number(5)/to_number(2) from dual;
 ?COLUMN? 
----------
      2.5
(1 row)

select to_number(5)/'2' from dual;
 ?COLUMN? 
----------
      2.5
(1 row)

select 1.0::binary_float/'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1.0::float(4)/'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1/'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1/'1.0' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select '1'/'1.0' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select '1'/'1' from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1/to_char(1) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select to_char(1)/to_char(1.2) from dual;
        ?COLUMN?        
------------------------
 0.83333333333333333333
(1 row)

select 1/to_char(1) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1/'1.0'::varchar2(3) from dual;
 ?COLUMN? 
----------
        1
(1 row)

select 1/'1'::char(3) from dual;
 ?COLUMN? 
----------
        1
(1 row)

--比较运算符	
create table signtbl1(id number, name varchar2(3));
insert into signtbl1 values(1,'1.0');
select * from signtbl1 where name = 1;
 ID | NAME 
----+------
  1 | 1.0
(1 row)

drop table signtbl1;
select 'equal' from dual where 1>=1.0::varchar(3);
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1>=to_char(1.0);
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1>='1.0';
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1>=1::varchar(3);
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1>=1::char(3);
 ?COLUMN? 
----------
 equal
(1 row)

create table signtbl1(id number,job varchar(10));
insert into signtbl1 values(1,'IT');
insert into signtbl1 values(2,'doctor');
insert into signtbl1 values(3,'teacher');
insert into signtbl1 values(6,'');
create table signtbl2(id char,avgm number);
insert into signtbl2 values(1.00,10000);
insert into signtbl2 values(2,5000);
insert into signtbl2 values(4,9000);
insert into signtbl2 values(5,null);
select * from signtbl1 where id > all(select id from signtbl2);
 ID | JOB 
----+-----
  6 | 
(1 row)

drop table signtbl1;
drop table signtbl2;
--between and		
select 1 from dual where 3 between to_char(2) and to_char(5);
 ?COLUMN? 
----------
        1
(1 row)

select 1 from dual where 3 between 2.0::char(5) and 5.0::varchar(5);
 ?COLUMN? 
----------
        1
(1 row)

select 1 from dual where 3 between '2.0' and '5.0';
 ?COLUMN? 
----------
        1
(1 row)

select 1 from dual where to_char(3) between '2.0' and to_char(5);
 ?COLUMN? 
----------
        1
(1 row)

select 1 from dual where to_char(3) between to_char(2) and to_char(5);
 ?COLUMN? 
----------
        1
(1 row)

select 1 from dual where 3.0 between '2.0' and '5.0';
 ?COLUMN? 
----------
        1
(1 row)

select 'equal' from dual where 1=1::number(2);
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1=1.0::number(3,2);
 ?COLUMN? 
----------
 equal
(1 row)

select 'equal' from dual where 1=1.0::binary_float;
 ?COLUMN? 
----------
 equal
(1 row)

--||	
select 1||2 as result from dual;
 RESULT 
--------
 12
(1 row)

select 1.0||2.0 as result from dual;
 RESULT 
--------
 12
(1 row)

select to_number(1.20)||to_number(2.0) as result from dual;
 RESULT 
--------
 1.22
(1 row)

select 1.00::binary_double||2.00::binary_double as result from dual;
 RESULT 
--------
 12
(1 row)

select 1||to_char('2') as result from dual;
 RESULT 
--------
 12
(1 row)

select 1.0||to_char('2') as result from dual;
 RESULT 
--------
 12
(1 row)

select to_char(1.0)||to_char(2.0) as result from dual;
 RESULT 
--------
 12
(1 row)

select to_char(1.00)||to_char('b') as result from dual;
 RESULT 
--------
 1b
(1 row)

select 1||'2' as result from dual;
 RESULT 
--------
 12
(1 row)

select '1'||'2' as result from dual;
 RESULT 
--------
 12
(1 row)

select to_char(1)||to_char(2) as result from dual;
 RESULT 
--------
 12
(1 row)

select to_char('1.0')||to_char('2.0') as result from dual;
 RESULT 
--------
 1.02.0
(1 row)

select to_char('a')||to_char('b') as result from dual;
 RESULT 
--------
 ab
(1 row)

select to_char(1)||to_char('b') as result from dual;
 RESULT 
--------
 1b
(1 row)

